server/services/frontendComponentService.ts

```diff
@@ -1,17 +1,26 @@
-import type { FrontendComponent } from 'frontendComponentApiClient'
+import type { FrontendComponentsPageAdditions } from 'viewModels'
 import { FrontendComponentApiClient } from '../data'
 import logger from '../../logger'
+import toFrontendComponentsPageAdditions from '../data/mappers/frontendComponentMapper'
 
 export default class FrontendComponentService {
   constructor(private readonly frontendApiClient: FrontendComponentApiClient) {}
 
-  async getComponents(component: 'header' | 'footer', userToken: string): Promise<FrontendComponent> {
+  async getComponents(userToken: string): Promise<FrontendComponentsPageAdditions> {
```

Comment:
This is the main refactoring in this PR in respect of design change ....
The service is now consistent with our other services in that it gets data from an API, maps it into a view model, and returns it. If there is an error it returns the declared type, but with the field problemRetrievingData set to true. This is what all our services do 👍 

The specifics of this service method are that it actually calls the frontend components API twice - once to get the header, once to get the footer; then maps both responses into the returned FrontendComponentsPageAdditions object